File: code/dart/lib/src/graphql/sqlserver/physical_model.dart
enum PhysicalObjectKind {
  table,
  view,
}

final class PhysicalCatalog {
  const PhysicalCatalog({required this.objects});

  final List<PhysicalObject> objects;
}

final class PhysicalObject {
  const PhysicalObject({
    required this.id,
    required this.kind,
    required this.schemaName,
    required this.objectName,
    required this.identityFields,
    required this.fields,
    required this.relations,
  });

  final String id;
  final PhysicalObjectKind kind;
  final String schemaName;
  final String objectName;
  final List<String> identityFields;
  final List<PhysicalField> fields;
  final List<PhysicalRelationSeed> relations;
}

final class PhysicalField {
  const PhysicalField({
    required this.column,
    required this.nativeType,
    required this.nullable,
  });

  final String column;
  final String nativeType;
  final bool nullable;
}

final class PhysicalRelationSeed {
  const PhysicalRelationSeed({
    required this.name,
    required this.sourceObjectId,
    required this.targetObjectId,
    required this.sourceFields,
    required this.targetFields,
  });

  final String name;
  final String sourceObjectId;
  final String targetObjectId;
  final List<String> sourceFields;
  final List<String> targetFields;
}

File: code/ts/modular_api/src/graphql/sqlserver/physical_model.ts
export enum PhysicalObjectKind {
  Table = 'table',
  View = 'view',
}

export interface PhysicalCatalog {
  readonly objects: readonly PhysicalObject[];
}

export interface PhysicalObject {
  readonly id: string;
  readonly kind: PhysicalObjectKind;
  readonly schemaName: string;
  readonly objectName: string;
  readonly identityFields: readonly string[];
  readonly fields: readonly PhysicalField[];
  readonly relations: readonly PhysicalRelationSeed[];
}

export interface PhysicalField {
  readonly column: string;
  readonly nativeType: string;
  readonly nullable: boolean;
}

export interface PhysicalRelationSeed {
  readonly name: string;
  readonly sourceObjectId: string;
  readonly targetObjectId: string;
  readonly sourceFields: readonly string[];
  readonly targetFields: readonly string[];
}